cmd/openshift-install/main.go

```diff
@@ -14,7 +18,13 @@ var (
 )
 
 func main() {
-	rootCmd := newRootCmd()
+	logfile, err := os.OpenFile(".openshift-install.log", os.O_WRONLY|os.O_CREATE, 0666)
```

Comment:
Hrm.  I'd rather not create this file as a result of `openshift-install help` or `version` calls.  Not sure what the best way to handle that is.  Can we put helpers in `cmd/openshift-install/log.go`, set up `PersistentPreRunE` for the `create` and `destroy` sub-commands, and then use a package-global `chan` to trigger the close (and block program exit until the close completes)?

And maybe we want to put this inside the assets directory (which we already require to be writeable)?  Is the leading period sufficient to get the asset tooling to ignore it?

Reply:
Some of the alternate hook implementations I looked at open and close the file every time they log. This seems like it would be slow and shouldn't really be necessary. On the other hand, it doesn't look like there is a good way to ensure that a file is flushed/closed when the program exits. Maybe there is a hook I can use within cobra to ensure the file is closed when I am through.

Reply:
> Put the file in a package global and close it from `PersistentPostRunE`?

It kills me that this is a reasonable approach, given the tools in hand.

Reply:
PersistentPostRunE will not run if any previous execs fail

Reply:
> PersistentPostRunE will not run if any previous execs fail

Oh, right 😢 .  Well, with a package global `*File`, we could put a `defer` in `main` to cloe if and only if the global was non-`nil`?

Reply:
Okay. @abhinavdahiya and I think we have a decent solution. Let me know what you think.